### PR TITLE
Kim improvement

### PR DESCRIPTION
**Pull request by yafshar**
**Summary**

This PR contains KIM updates, including:

- Update the eflag_global & eflag_atom  
- Extra unit tests, (following https://github.com/lammps/lammps/pull/2477#discussion_r531143686 discussion) KIM_EXTRA_UNITTESTS option is added to `cmake/Modules/Packages/KIM.cmake` which defaults to OFF. The CMake variable and all the requirements are documented in `doc/src/Build_extras.rst`.

**Related Issue(s)**

No related issues.

**Author(s)**

Yaser Afshar @yafshar (UMN)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

The pull request is backward compatible.

**Implementation Notes**

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

File: unittest/commands/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,13 @@ target_link_libraries(test_lattice_region PRIVATE lammps GTest::GMock GTest::GTe
 add_test(NAME LatticeRegion COMMAND test_lattice_region WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
 
 add_executable(test_kim_commands test_kim_commands.cpp)
+if(KIM_EXTRA_UNITTESTS)
+  if(CURL_FOUND)
+    target_compile_definitions(test_kim_commands PRIVATE -DKIM_EXTRA_UNITTESTS)
+  else()
+    message(STATUS "CURL not found. Enabling KIM extra unit tests requires to have libcurl installed.")
```
Comment (akohlmey) [inline unittest/commands/CMakeLists.txt:15]: shouldn't this be a `FATAL_ERROR` instead of a `STATUS` message?

Comment (yafshar) [reply unittest/commands/CMakeLists.txt]: thanks, yes it makes sense. I updated the mode to `FATAL_ERROR`.

**Review by sjplimp — APPROVED** [review]

**Comment by akohlmey** [comment]
@yafshar just merged this change and then I had an idea of a further improvement. 
- the test command(s) could use a virtual environment in the build folder where all required python packages are installed
- the CMake script could do the installation of KIM models as part of the build, too.
that would make the process even more automatic (and thus more likely to be used).

**Comment by yafshar** [comment]
> @yafshar just merged this change and then I had an idea of a further improvement.
> 
>     * the test command(s) could use a virtual environment in the build folder where all required python packages are installed
>     * the CMake script could do the installation of KIM models as part of the build, too.
>       that would make the process even more automatic (and thus more likely to be used).
@akohlmey sounds great. I can help if you want to do this.

**Pull request merged**